spawn.lua

```diff
@@ -1,66 +1,70 @@
+	PermissionMissingSpecific = "You need %s permission to do that!",
+	SpawnReached = "Returned to %s's spawn",
+	SpawnSet = "Changed spawn position of %s to [X:%i Y:%i Z:%i]",
+	SpawnSetFailure = "Couldn't change spawn position of %s to [X:%i Y:%i Z:%i]"
+}
+
+local function SendToWorldSpawn( Player, Callback )
+
+	local World = Player:GetWorld()
+	local Chunks = {}
+	table.insert( Chunks, { math.floor(World:GetSpawnX() / 16), math.floor(World:GetSpawnZ() / 16) } )
+	local OnAllChunksAvailable = function()
+		Player:TeleportToCoords(World:GetSpawnX(), World:GetSpawnY(), World:GetSpawnZ())
+		Callback()
+	end -- OnAllChunksAvailable
+	World:ChunkStay(Chunks, nil, OnAllChunksAvailable)
+
+end
+
 function HandleSpawnCommand(Split, Player)
 	
-	local WorldIni = cIniFile()
-	WorldIni:ReadFile(Player:GetWorld():GetIniFileName())
-	
-	local SpawnX = WorldIni:GetValue("SpawnPosition", "X")
-	local SpawnY = WorldIni:GetValue("SpawnPosition", "Y")
-	local SpawnZ = WorldIni:GetValue("SpawnPosition", "Z")
-	local flag = 0
-	
-	if (#Split == 2 and Split[2] ~= Player:GetName()) then
+	if ( (#Split == 2) and (Split[2] ~= Player:GetName()) ) then
 		if Player:HasPermission("core.spawn.others") then
+			local IsPlayerFound
 			local FoundPlayerCallback = function(OtherPlayer)
 				if (OtherPlayer:GetName() == Split[2]) then
-					World = OtherPlayer:GetWorld()
-					local OnAllChunksAvaliable = function()
-						OtherPlayer:TeleportToCoords(SpawnX, SpawnY, SpawnZ)
-						SendMessageSuccess( Player, "Returned " .. OtherPlayer:GetName() .. " to world spawn" )
-						flag=1
-					end
-					World:ChunkStay({{SpawnX/16, SpawnZ/16}}, OnChunkAvailable, OnAllChunksAvaliable)		
+					OtherPlayer:SendMessageInfo( string.format( Str.SpawnBeingSend, OtherPlayer:GetWorld():GetName() ) )
+					local Feedback = function()
+						Player:SendMessageSuccess( string.format( Str.SpawnSentOther, OtherPlayer:GetName(), OtherPlayer:GetWorld():GetName() ) )
+					end -- Feedback
+					SendToWorldSpawn( OtherPlayer, Feedback )
+					IsPlayerFound = true
 				end
 			end
 			cRoot:Get():FindAndDoWithPlayer(Split[2], FoundPlayerCallback)
 			
-			if flag == 0 then
-				SendMessageFailure( Player, "Player " .. Split[2] .. " not found!" )
+			if not(IsPlayerFound) then
+				Player:SendMessageFailure( string.format( Str.PlayerNotFound, Split[2]) )
 			end
 		else
-			SendMessageFailure( Player, "You need core.spawn.others permission to do that!" )
+			Player:SendMessageFailure( string.format( Str.PermissionMissingSpecific, "core.spawn.others" ) )
 		end
 	else
-		World = Player:GetWorld()
-		local OnAllChunksAvaliable = function()
-			Player:TeleportToCoords(SpawnX, SpawnY, SpawnZ)
-			SendMessageSuccess( Player, "Returned to world spawn" )
-		end
-		World:ChunkStay({{SpawnX/16, SpawnZ/16}}, OnChunkAvailable, OnAllChunksAvaliable)
+		local Feedback = function()
+			Player:SendMessageInfo( string.format( Str.SpawnReached, Player:GetWorld():GetName() ) )
```

Comment:
Again, cannot use `Player` in ChunkStay callback.
(This applies to all `Feedback` functions here)